### PR TITLE
ref guide warnings (closes #916)

### LINKED ISSUE #916: Warnings when building Reference Guide
When building the Reference guide we get a lot (25) of warnings about incorrectly formatted docstrings.

### PR DESCRIPTION
**Pull request by arporter**
A small PR that fixes formatting errors in the docstrings.

### CODE REVIEW

File: doc/reference_guide/source/conf.py [hunk 1]
```diff
@@ -21,13 +21,13 @@
...
 project = 'Reference Guide'
-copyright = '2019, R. W. Ford, J. Henrichs, I. Kavcic, A. R. Porter, S. Siso'
+copyright = '2019-2020, R. W. Ford, J. Henrichs, I. Kavcic, A. R. Porter, S. Siso'
 author = 'R. W. Ford, J. Henrichs, I. Kavcic, A. R. Porter, S. Siso'
 
 # The short X.Y version
 version = ''
 # The full version, including alpha/beta/rc tags
-release = '1.7.0'
+release = '1.9.1'
```
Comment (rupertford) [inline doc/reference_guide/source/conf.py:30]: The wiki talks about updating doxygen.config on a release. Would that update this file, or does this need to be done manually as well? If so, it should be added to the wiki. And something to say to check the copyright year too.

File: src/psyclone/dynamo0p3.py [hunk 2]
```diff
@@ -5702,25 +5702,19 @@ def __init__(self):
...
     @property
     def annexed_only(self):
-        '''Returns whether the access to the halo is solely to annexed dofs,
-        or not
-
-        :return: Return True if only annexed dofs are accessed in the
-        halo and False otherwise
+        '''
+        :returns: True if only annexed dofs are accessed in the halo and \
+                  False otherwise
```
Comment (rupertford) [inline src/psyclone/dynamo0p3.py:5707]: Full stop please.

File: src/psyclone/dynamo0p3.py [hunk 3]
```diff
@@ -5702,25 +5702,19 @@ def __init__(self):
...
     @property
     def max_depth(self):
-        '''Returns whether the read to the field is known to access all of the
-        halo or not
-
-        :return: Return True if the read to the field is known to
-        access all of the halo and False otherwise
+        '''
+        :returns: True if the read to the field is known to access all \
+                  of the halo and False otherwise
```
Comment (rupertford) [inline src/psyclone/dynamo0p3.py:5716]: full stop please.

File: src/psyclone/dynamo0p3.py [hunk 4]
```diff
@@ -5729,8 +5723,8 @@ def max_depth_m1(self):
         '''Returns whether the read to the field is known to access all of the
         halo except the outermost level or not.
 
-        :return: Return True if the read to the field is known to
-        access all of the halo except the outermost and False otherwise
+        :returns: True if the read to the field is known to access all \
+                  of the halo except the outermost and False otherwise
```
Comment (rupertford) [inline src/psyclone/dynamo0p3.py:5727]: Full stop please.

File: src/psyclone/dynamo0p3.py [hunk 5]
```diff
@@ -5743,9 +5737,9 @@ def var_depth(self):
         stencil accesses. Also note, this depth should be added to the
         literal_depth to find the total depth.
 
-        :return: Return a variable name specifying the halo
-        access depth, if one exists, and None if not
-        :rtype: String
+        :returns: a variable name specifying the halo access depth \
+                  if one exists, and None if not
```
Comment (rupertford) [inline src/psyclone/dynamo0p3.py:5741]: full stop please.

File: src/psyclone/dynamo0p3.py [hunk 6]
```diff
@@ -5756,9 +5750,8 @@ def literal_depth(self):
         depth should be added to the var_depth to find the total
         depth.
 
-        :return: Return the known fixed (literal) halo
-        access depth
-        :rtype: integer
+        :returns: Return the known fixed (literal) halo access depth.
```
Comment (rupertford) [inline src/psyclone/dynamo0p3.py:5753]: Remove "Return"

File: src/psyclone/dynamo0p3.py [hunk 7]
```diff
@@ -5884,8 +5876,8 @@ def dirty_outer(self):
         been written to is actually dirty (well to be precise it is a partial
         sum).
 
-        :return: Return True if the outer layer of halo
-        that is written to remains dirty and False otherwise.
+        :return: True if the outer layer of halo that is written \
```
Comment (rupertford) [inline src/psyclone/dynamo0p3.py:5879]: return -> returns (for consistency)

File: src/psyclone/dynamo0p3.py [hunk 8]
```diff
@@ -6293,12 +6285,12 @@ def upper_bound_name(self):
...
     def upper_bound_halo_depth(self):
         '''Returns the index of the upper loop bound. This is None if the upper
-        bound name is not in HALO_ACCESS_LOOP_BOUNDS
+        bound name is not in HALO_ACCESS_LOOP_BOUNDS.
 
-        :return: the depth of the halo for a loops upper bound. If it
-        is None then a depth has not been provided. The depth value is only
-        valid when the upper-bound name is associated with a halo
-        e.g. 'cell_halo'
+        :return: the depth of the halo for a loops upper bound. If it \
```
Comment (rupertford) [inline src/psyclone/dynamo0p3.py:6290]: return -> returns (for consistency)

File: src/psyclone/psyir/nodes/node.py [hunk 9]
```diff
@@ -89,7 +92,7 @@ def colored(text, _):
 
         :param text: Text to return
         :type text: string
-        :param _: Fake argument, only required to match interface
+        :param _: Fake argument, only required to match interface \
```
Comment (rupertford) [inline src/psyclone/psyir/nodes/node.py:95]: Full stop at end of sentence.

Comment (rupertford) [reply src/psyclone/psyir/nodes/node.py:95]: Perhaps fix the above param as well (lower case and full stop).

**Review by rupertford — CHANGES_REQUESTED** [review]
Just a few issues with full stops and consistency. Please see inline comments.

File: doc/reference_guide/source/conf.py (hunk 1 above)
Comment (arporter) [reply doc/reference_guide/source/conf.py:30]: Oh yes, done.

File: src/psyclone/dynamo0p3.py (hunk 2 above)
Comment (arporter) [reply src/psyclone/dynamo0p3.py:5707]: Done.

File: src/psyclone/dynamo0p3.py (hunk 3 above)
Comment (arporter) [reply src/psyclone/dynamo0p3.py:5716]: Done.

File: src/psyclone/dynamo0p3.py (hunk 4 above)
Comment (arporter) [reply src/psyclone/dynamo0p3.py:5727]: Done.

File: src/psyclone/dynamo0p3.py (hunk 5 above)
Comment (arporter) [reply src/psyclone/dynamo0p3.py:5741]: Done.

File: src/psyclone/dynamo0p3.py (hunk 6 above)
Comment (arporter) [reply src/psyclone/dynamo0p3.py:5753]: Done.

File: src/psyclone/dynamo0p3.py (hunk 7 above)
Comment (arporter) [reply src/psyclone/dynamo0p3.py:5879]: Done.

File: src/psyclone/dynamo0p3.py (hunk 8 above)
Comment (arporter) [reply src/psyclone/dynamo0p3.py:6290]: Done.

File: src/psyclone/psyir/nodes/node.py (hunk 9 above)
Comment (arporter) [reply src/psyclone/psyir/nodes/node.py:95]: Done.

**Comment by arporter** [comment]
I noticed that one of my docstring fixes was now causing a pycodestyle warning so I've reformatted it to use in-line literal expressions instead. Assuming Travis is happy this is ready for next review.

File: src/psyclone/psyir/nodes/node.py (hunk 9 above)
Comment (rupertford) [reply src/psyclone/psyir/nodes/node.py:94]: There is an uppercase issue here but I've taken the liberty to fix.

File: src/psyclone/domain/lfric/lfric_arg_descriptor.py [hunk 10]
```diff
@@ -738,7 +738,7 @@ def function_spaces(self):
     @property
     def vector_size(self):
         '''
-        Returns the vector size of the argument. This will be 1 if *n
+        Returns the vector size of the argument. This will be 1 if '\*n'
```
Comment (rupertford) [inline src/psyclone/domain/lfric/lfric_arg_descriptor.py:741]: pycodestyle complains here ...

lfric_arg_descriptor.py:741:69: W605 invalid escape sequence '\*'

Presumably pycodestyle does this in error seeing as we're in a string?

File: doc/reference_guide/source/conf.py [hunk 11]
```diff
@@ -21,13 +21,13 @@
 # -- Project information -----------------------------------------------------
 
 project = 'Reference Guide'
-copyright = '2019, R. W. Ford, J. Henrichs, I. Kavcic, A. R. Porter, S. Siso'
+copyright = '2019-2020, R. W. Ford, J. Henrichs, I. Kavcic, A. R. Porter, S. Siso'
```
Comment (rupertford) [inline doc/reference_guide/source/conf.py:24]: pycodestyle says line is too long. I'll not ask for pylint compliance for the file seeing as this is a small modification.

**Review by rupertford — CHANGES_REQUESTED** [review]
Just a couple of pycodestyle issues - sorry.

File: src/psyclone/domain/lfric/lfric_arg_descriptor.py (hunk 10 above)
Comment (arporter) [reply src/psyclone/domain/lfric/lfric_arg_descriptor.py:741]: Yes, I thought it should be an error but apparently it isn't. I've found a better solution now.

File: doc/reference_guide/source/conf.py (hunk 11 above)
Comment (arporter) [reply doc/reference_guide/source/conf.py:24]: Oh yes, fixed.

**Review by rupertford — CHANGES_REQUESTED** [review]
All changes look good. However, when I build the reference guide I get the following warnings ...

`
/home/rupert/proj/PSyclone/src/psyclone/psyir/nodes/__init__.py:docstring of psyclone.psyir.nodes.SCHEDULE_COLOUR_MAP:3: WARNING: Unexpected indentation.                                                                                                       /home/rupert/proj/PSyclone/src/psyclone/psyir/nodes/__init__.py:docstring of psyclone.psyir.nodes.SCHEDULE_COLOUR_MAP:4: WARNING: Block quote ends without a blank line; unexpected unindent.                                                                   /home/rupert/proj/PSyclone/src/psyclone/psyir/nodes/__init__.py:docstring of psyclone.psyir.nodes.SCHEDULE_COLOUR_MAP:7: WARNING: Unexpected indentation.                                                                                                       /home/rupert/proj/PSyclone/src/psyclone/psyir/nodes/__init__.py:docstring of psyclone.psyir.nodes.SCHEDULE_COLOUR_MAP:9: WARNING: Inline strong start-string without end-string.                                                                                /home/rupert/proj/PSyclone/src/psyclone/psyir/symbols/__init__.py:docstring of psyclone.psyir.symbols.TYPE_MAP_TO_PYTHON:3: WARNING: Unexpected indentation.                                                                                                    /home/rupert/proj/PSyclone/src/psyclone/psyir/symbols/__init__.py:docstring of psyclone.psyir.symbols.TYPE_MAP_TO_PYTHON:4: WARNING: Block quote ends without a blank line; unexpected unindent.                                                                /home/rupert/proj/PSyclone/src/psyclone/psyir/symbols/__init__.py:docstring of psyclone.psyir.symbols.TYPE_MAP_TO_PYTHON:7: WARNING: Unexpected indentation.                                                                                                    /home/rupert/proj/PSyclone/src/psyclone/psyir/symbols/__init__.py:docstring of psyclone.psyir.symbols.TYPE_MAP_TO_PYTHON:9: WARNING: Inline strong start-string without end-string.
`

Could you please take a look.

**Comment by arporter** [comment]
Yes, those warnings are to do with the fact that sphinx/autodoc is pulling in the documentation for the `dict` class and those errors are in the doc strings for that class. (I've noted this in #916.) I therefore cannot do anything about them.
I've brought the branch up-to-date with master.

**Comment by arporter** [comment]
I've posted about this here: https://groups.google.com/g/sphinx-users/c/ayjm7k--xZI
I'll set this as 'in progress' while I wait to see whether I get an answer...

**Comment by rupertford** [comment]
I don't think this PR should have to wait on an external bug. We can keep the issue open.

**Review by rupertford — APPROVED** [review]
All issues have been addressed. As discussed there are some remaining warnings but these are due to doc strings in an external library.

**Pull request merged**